Plugins directory mounting issue

Hi,

I'm trying to install plugins through this helm chart using "9.2.7" version however i see an issue when the ephemeral volume is used. I get the below error 

```
java.lang.IllegalStateException: Fail to move plugin: /opt/sonarqube/extensions/downloads/sonar-findbugs-plugin-4.0.3.jar to /opt/sonarqube/extensions/plugins/sonar-findbugs-plugin-4.0.3.jar
	at org.sonar.server.plugins.ServerPluginRepository.overrideAndRegisterPlugin(ServerPluginRepository.java:185)
	at org.sonar.server.plugins.ServerPluginRepository.moveDownloadedPlugins(ServerPluginRepository.java:146)
	at org.sonar.server.plugins.ServerPluginRepository.start(ServerPluginRepository.java:104)
	at org.sonar.core.platform.StartableCloseableSafeLifecyleStrategy.start(StartableCloseableSafeLifecyleStrategy.java:40)
	at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.start(AbstractInjectionFactory.java:84)
	at org.picocontainer.behaviors.AbstractBehavior.start(AbstractBehavior.java:169)
	at org.picocontainer.behaviors.Stored$RealComponentLifecycle.start(Stored.java:132)
	at org.picocontainer.behaviors.Stored.start(Stored.java:110)
	at org.picocontainer.DefaultPicoContainer.potentiallyStartAdapter(DefaultPicoContainer.java:1016)
	at org.picocontainer.DefaultPicoContainer.startAdapters(DefaultPicoContainer.java:1009)
	at org.picocontainer.DefaultPicoContainer.start(DefaultPicoContainer.java:767)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.server.platform.platformlevel.PlatformLevel.start(PlatformLevel.java:90)
	at org.sonar.server.platform.platformlevel.PlatformLevel2.start(PlatformLevel2.java:113)
	at org.sonar.server.platform.Platform.start(Platform.java:211)
	at org.sonar.server.platform.Platform.startLevel2Container(Platform.java:177)
	at org.sonar.server.platform.Platform.init(Platform.java:87)
	at org.sonar.server.platform.web.PlatformServletContextListener.contextInitialized(PlatformServletContextListener.java:43)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4817)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5283)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1423)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1413)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: Destination '/opt/sonarqube/extensions/plugins' directory cannot be created
	at org.apache.commons.io.FileUtils.copyFile(FileUtils.java:1070)
	at org.apache.commons.io.FileUtils.copyFile(FileUtils.java:1028)
	at org.apache.commons.io.FileUtils.moveFile(FileUtils.java:3008)
	at org.sonar.server.plugins.ServerPluginRepository.overrideAndRegisterPlugin(ServerPluginRepository.java:182)
	... 26 common frames omitted
```

Whereas if i use lower versions like 9.2.6, 9.2.5 i get permission denied error, PFB stacktrace
`java.lang.IllegalStateException: Fail to move plugin: /opt/sonarqube/extensions/downloads/sonar-findbugs-plugin-4.0.3.jar to /opt/sonarqube/extensions/plugins/sonar-findbugs-plugin-4.0.3.jar
	at org.sonar.server.plugins.ServerPluginRepository.overrideAndRegisterPlugin(ServerPluginRepository.java:185)
	at org.sonar.server.plugins.ServerPluginRepository.moveDownloadedPlugins(ServerPluginRepository.java:146)
	at org.sonar.server.plugins.ServerPluginRepository.start(ServerPluginRepository.java:104)
	at org.sonar.core.platform.StartableCloseableSafeLifecyleStrategy.start(StartableCloseableSafeLifecyleStrategy.java:40)
	at org.picocontainer.injectors.AbstractInjectionFactory$LifecycleAdapter.start(AbstractInjectionFactory.java:84)
	at org.picocontainer.behaviors.AbstractBehavior.start(AbstractBehavior.java:169)
	at org.picocontainer.behaviors.Stored$RealComponentLifecycle.start(Stored.java:132)
	at org.picocontainer.behaviors.Stored.start(Stored.java:110)
	at org.picocontainer.DefaultPicoContainer.potentiallyStartAdapter(DefaultPicoContainer.java:1016)
	at org.picocontainer.DefaultPicoContainer.startAdapters(DefaultPicoContainer.java:1009)
	at org.picocontainer.DefaultPicoContainer.start(DefaultPicoContainer.java:767)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:135)
	at org.sonar.server.platform.platformlevel.PlatformLevel.start(PlatformLevel.java:90)
	at org.sonar.server.platform.platformlevel.PlatformLevel2.start(PlatformLevel2.java:113)
	at org.sonar.server.platform.Platform.start(Platform.java:211)
	at org.sonar.server.platform.Platform.startLevel2Container(Platform.java:177)
	at org.sonar.server.platform.Platform.init(Platform.java:87)
	at org.sonar.server.platform.web.PlatformServletContextListener.contextInitialized(PlatformServletContextListener.java:43)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4817)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5283)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1423)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1413)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.FileNotFoundException: /opt/sonarqube/extensions/plugins/sonar-findbugs-plugin-4.0.3.jar (Permission denied)
	at java.base/java.io.FileOutputStream.open0(Native Method)
	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:298)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:237)
	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:187)
	at org.apache.commons.io.FileUtils.doCopyFile(FileUtils.java:1123)
	at org.apache.commons.io.FileUtils.copyFile(FileUtils.java:1076)
	at org.apache.commons.io.FileUtils.copyFile(FileUtils.java:1028)
	at org.apache.commons.io.FileUtils.moveFile(FileUtils.java:3008)
	at org.sonar.server.plugins.ServerPluginRepository.overrideAndRegisterPlugin(ServerPluginRepository.java:182)
	... 26 common frames omitted`

To solve this, i had to enable persistence volume by mounting "/opt/sonarqube/extensions" and use version 9.2.5. Version > 9.2.5 doesn't work with persistence enabled as well.

--- a/charts/sonarqube/Chart.yaml
+++ b/charts/sonarqube/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: sonarqube
 description: SonarQube is an open sourced code quality scanning tool
-version: 9.3.0
+version: 9.3.1
 appVersion: 8.5.1-community
 keywords:
   - coverage

--- a/charts/sonarqube/templates/deployment.yaml
+++ b/charts/sonarqube/templates/deployment.yaml
@@ -266,9 +266,15 @@ spec:
             - mountPath: {{ .Values.sonarqubeFolder }}/data
               name: sonarqube
               subPath: data
+            {{- if .Values.persistence.enabled }}
             - mountPath: {{ .Values.sonarqubeFolder }}/extensions
               name: sonarqube
               subPath: extensions
+            {{- else if .Values.plugins.install }}
+            - mountPath: {{ .Values.sonarqubeFolder }}/extensions/downloads
+              name: sonarqube
+              subPath: extensions/downloads
+            {{- end }}
             {{- if .Values.plugins.lib }}
             {{- range $index, $val := .Values.plugins.lib }}
             - mountPath: {{ $.Values.sonarqubeFolder }}/lib/common/{{ $val }}

